controllers/argocd/server/server.go

```diff
@@ -20,6 +24,112 @@ type ServerReconciler struct {
 
 func (sr *ServerReconciler) Reconcile() error {
 
-	// controller logic goes here
+	sr.Logger = ctrl.Log.WithName(ServerControllerComponent).WithValues("instance", sr.Instance.Name, "instance-namespace", sr.Instance.Namespace)
+
+	// namespace doesn't exist, break reconciliation
+	ns, err := cluster.GetNamespace(sr.Instance.Namespace, sr.Client)
+	if err != nil {
+		sr.Logger.Error(err, "Reconcile: failed to retrieve namespace", "name", sr.Instance.Namespace)
+		return err
+	}
+
+	// namespace is deleting, delete the resources
+	if ns.DeletionTimestamp != nil {
+		return sr.DeleteResources()
+	}
```

Comment:
I think this is no longer necessary

We will add some logic to the namespace watcher's event handler to watch for terminating namespaces and trigger deletion of all controllers' resources from there instead of running this check each time in every controller unnecessarily